src/peripheral/dwt.rs

```diff
@@ -58,16 +69,57 @@ pub struct Comparator {
     /// Comparator Mask
     pub mask: RW<u32>,
     /// Comparator Function
-    pub function: RW<u32>,
+    pub function: RW<Function>,
     reserved: u32,
 }
 
+bitfield! {
+    #[repr(C)]
+    #[derive(Copy, Clone)]
+    /// Comparator FUNCTIONn register.
+    pub struct Function(u32);
+    u8, get_function, set_function: 3, 0;
+    get_emitrange, set_emitrange: 5;
+    get_cycmatch, set_cycmatch: 7;
+    get_datavmatch, set_datavmatch: 8;
+    get_matched, _: 24;
+}
+
 impl DWT {
     /// Enables the cycle counter
     #[cfg(not(armv6m))]
     #[inline]
     pub fn enable_cycle_counter(&mut self) {
-        unsafe { self.ctrl.modify(|r| r | 1) }
+        unsafe {
+            self.ctrl.modify(|mut r| {
+                r.set_cyccntena(true);
+                r
+            });
+        }
+    }
+
+    /// Whether to enable exception tracing
+    // TODO find out if this is supported om armv6m
```

Comment:
I don't believe this is supported on ARMv6M: the architecture reference manual says of the DWT "This provides watchpoint support. Trace is not supported." and indeed there's no EXCTRCENA bit in DWT_CTRL.

Reply:
Both functions gated from `armv6m`.